// angular-reactive-form/esm5/lib/directives/custom-switch-input/custom-switch-input.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input, ElementRef, Renderer } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var CustomSwitchInputComponent = /** @class */ (function () {
    function CustomSwitchInputComponent(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
        this.onTouched = (/**
         * @return {?}
         */
        function () { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.writeValue = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        if (val) {
            this.value = val;
        }
    };
    Object.defineProperty(CustomSwitchInputComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.onChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.setDisabledState = /**
     * @param {?} isDisabled
     * @return {?}
     */
    function (isDisabled) {
        this.disabled = isDisabled;
    };
    /**
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.isClear === true) {
            setTimeout((/**
             * @return {?}
             */
            function () {
                _this.value = false;
            }), 1);
        }
    };
    /**
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            if (!_this.value || _this.value === '') {
                _this.value = false;
            }
        }), 1);
    };
    /**
     * @param {?} values
     * @return {?}
     */
    CustomSwitchInputComponent.prototype.FieldsChange = /**
     * @param {?} values
     * @return {?}
     */
    function (values) {
        this.value = values.currentTarget.checked;
    };
    CustomSwitchInputComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'custom-switch-input',
                    template: "<!-- <input type=\"checkbox\" [(ngModel)]=\"value\" id=\"switch\" (change)=\"FieldsChange($event)\" /><label for=\"switch\">Toggle</label> -->\n<div class=\"flatRoundedCheckbox\">\n  <input\n    [disabled]=\"isDisabled\"\n    type=\"checkbox\"\n    [(ngModel)]=\"value\"\n    id=\"rf_switch-{{ id }}\"\n    (change)=\"FieldsChange($event)\"\n    type=\"checkbox\"\n    name=\"\"\n    id=\"t2\"\n    checked\n  />\n  <label for=\"rf_switch-{{ id }}\"></label>\n  <div></div>\n</div>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return CustomSwitchInputComponent; })),
                            multi: true
                        }
                    ],
                    styles: [".flatRoundedCheckbox{width:25px;height:11px;position:relative}.flatRoundedCheckbox div{width:100%;height:100%;background:#7b7b7b;border-radius:50px;position:relative;top:-21px}.flatRoundedCheckbox label{display:block;width:13px;height:13px;border-radius:50px;transition:.5s;cursor:pointer;position:absolute;top:2px;z-index:1;left:-1px;background:#fff;box-shadow:0 3px 6px 0 #00000029}.flatRoundedCheckbox input[type=checkbox]{width:0;height:0;opacity:0}.flatRoundedCheckbox input[type=checkbox]:checked~div{background:#f48131}.flatRoundedCheckbox input[type=checkbox]:checked~label{left:15px}"]
                }] }
    ];
    /** @nocollapse */
    CustomSwitchInputComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer }
    ]; };
    CustomSwitchInputComponent.propDecorators = {
        _value: [{ type: Input }],
        isClear: [{ type: Input }],
        id: [{ type: Input }],
        isDisabled: [{ type: Input }]
    };
    return CustomSwitchInputComponent;
}());
export { CustomSwitchInputComponent };
if (false) {
    /** @type {?} */
    CustomSwitchInputComponent.prototype._value;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.isClear;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.id;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.isDisabled;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.renderer;
}
