// angular-reactive-form/esm2015/lib/models/SelectionByApi.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class SelectionByApi extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'selection-by-api';
        this.resultTransformer = options['resultTransformer'] || null;
        this.apiUpload = options['apiUpload'] || '';
        this.headers = options['headers'] || null;
        this.fieldName = options['fieldName'] || null;
        this.multiple = options['multiple'] || false;
        this.lengthToSearch = options['lengthToSearch'] || 3;
        this.method = options['method'] || 'get';
    }
}
if (false) {
    /** @type {?} */
    SelectionByApi.prototype.controlType;
    /** @type {?} */
    SelectionByApi.prototype.resultTransformer;
    /** @type {?} */
    SelectionByApi.prototype.headers;
    /** @type {?} */
    SelectionByApi.prototype.apiUpload;
    /** @type {?} */
    SelectionByApi.prototype.fieldName;
    /** @type {?} */
    SelectionByApi.prototype.multiple;
    /** @type {?} */
    SelectionByApi.prototype.lengthToSearch;
    /** @type {?} */
    SelectionByApi.prototype.method;
}
