// angular-reactive-form/esm2015/lib/directives/custom-switch-input/custom-switch-input.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input, ElementRef, Renderer } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export class CustomSwitchInputComponent {
    /**
     * @param {?} element
     * @param {?} renderer
     */
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
        this.onTouched = (/**
         * @return {?}
         */
        () => { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    writeValue(val) {
        if (val) {
            this.value = val;
        }
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.onChange(this._value);
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        if (this.isClear === true) {
            setTimeout((/**
             * @return {?}
             */
            () => {
                this.value = false;
            }), 1);
        }
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (!this.value || this.value === '') {
                this.value = false;
            }
        }), 1);
    }
    /**
     * @param {?} values
     * @return {?}
     */
    FieldsChange(values) {
        this.value = values.currentTarget.checked;
    }
}
CustomSwitchInputComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'custom-switch-input',
                template: "<!-- <input type=\"checkbox\" [(ngModel)]=\"value\" id=\"switch\" (change)=\"FieldsChange($event)\" /><label for=\"switch\">Toggle</label> -->\n<div class=\"flatRoundedCheckbox\">\n  <input\n    [disabled]=\"isDisabled\"\n    type=\"checkbox\"\n    [(ngModel)]=\"value\"\n    id=\"rf_switch-{{ id }}\"\n    (change)=\"FieldsChange($event)\"\n    type=\"checkbox\"\n    name=\"\"\n    id=\"t2\"\n    checked\n  />\n  <label for=\"rf_switch-{{ id }}\"></label>\n  <div></div>\n</div>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => CustomSwitchInputComponent)),
                        multi: true
                    }
                ],
                styles: [".flatRoundedCheckbox{width:25px;height:11px;position:relative}.flatRoundedCheckbox div{width:100%;height:100%;background:#7b7b7b;border-radius:50px;position:relative;top:-21px}.flatRoundedCheckbox label{display:block;width:13px;height:13px;border-radius:50px;transition:.5s;cursor:pointer;position:absolute;top:2px;z-index:1;left:-1px;background:#fff;box-shadow:0 3px 6px 0 #00000029}.flatRoundedCheckbox input[type=checkbox]{width:0;height:0;opacity:0}.flatRoundedCheckbox input[type=checkbox]:checked~div{background:#f48131}.flatRoundedCheckbox input[type=checkbox]:checked~label{left:15px}"]
            }] }
];
/** @nocollapse */
CustomSwitchInputComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer }
];
CustomSwitchInputComponent.propDecorators = {
    _value: [{ type: Input }],
    isClear: [{ type: Input }],
    id: [{ type: Input }],
    isDisabled: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    CustomSwitchInputComponent.prototype._value;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.isClear;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.id;
    /** @type {?} */
    CustomSwitchInputComponent.prototype.isDisabled;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    CustomSwitchInputComponent.prototype.renderer;
}
