// angular-reactive-form/esm2015/lib/directives/mask-input/mask-input.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef, ElementRef, Renderer } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export class MaskInputComponent {
    /**
     * @param {?} element
     * @param {?} renderer
     */
    constructor(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.myModel = '';
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
        this.onTouched = (/**
         * @return {?}
         */
        () => { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    writeValue(val) {
        if (val) {
            this.value = val;
        }
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.onChange(this._value);
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @return {?}
     */
    ngOnChanges() {
        if (this.isClear === true) {
            setTimeout((/**
             * @return {?}
             */
            () => {
                if (this.value) {
                    this.value = null;
                    this.myModel = this.value;
                }
            }), 1);
        }
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (this.value) {
                this.myModel = this.value;
            }
        }), 1);
    }
    /**
     * @return {?}
     */
    change() {
        if (this.valueWithCharacter === true) {
            // tslint:disable-next-line:semicolon
            this.value = this.myModel;
        }
        else {
            this.value = this.myModel.replace(/\D+/g, '');
        }
    }
}
MaskInputComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'mask-input',
                template: "<input [attr.disabled]=\"isDisabled\" class=\"form-control\" placeholder=\"{{placeholder}}\" [textMask]=\"{mask: inputMaskText, guide: guide}\" (change)=\"change()\" [(ngModel)]=\"myModel\" type=\"text\"/>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => MaskInputComponent)),
                        multi: true
                    }
                ],
                styles: [""]
            }] }
];
/** @nocollapse */
MaskInputComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer }
];
MaskInputComponent.propDecorators = {
    inputMaskText: [{ type: Input }],
    guide: [{ type: Input }],
    _value: [{ type: Input }],
    valueWithCharacter: [{ type: Input }],
    placeholder: [{ type: Input }],
    isClear: [{ type: Input }],
    isDisabled: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    MaskInputComponent.prototype.inputMaskText;
    /** @type {?} */
    MaskInputComponent.prototype.guide;
    /** @type {?} */
    MaskInputComponent.prototype._value;
    /** @type {?} */
    MaskInputComponent.prototype.valueWithCharacter;
    /** @type {?} */
    MaskInputComponent.prototype.placeholder;
    /** @type {?} */
    MaskInputComponent.prototype.isClear;
    /** @type {?} */
    MaskInputComponent.prototype.isDisabled;
    /** @type {?} */
    MaskInputComponent.prototype.myModel;
    /** @type {?} */
    MaskInputComponent.prototype.selected;
    /**
     * @type {?}
     * @private
     */
    MaskInputComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    MaskInputComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    MaskInputComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    MaskInputComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    MaskInputComponent.prototype.renderer;
}
