// angular-reactive-form/esm5/lib/angular-reactive-form.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, Injector } from '@angular/core';
import { AngularReactiveFormComponent } from './angular-reactive-form.component';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { DirectivesModule } from './directives/directives.module';
import { EditorModule } from '@tinymce/tinymce-angular';
import { ReactiveFormComponent } from './components/reactive-form/reactive-form.component';
import { DynamicInputComponent } from './components/dynamic-input/dynamic-input.component';
import { setAppInjector } from './app-injector';
/** @type {?} */
export var UPLOAD_DIRECTIVE_HEADERS = 'UPLOAD_DIRECTIVE_HEADERS';
/** @type {?} */
export var UPLOAD_DIRECTIVE_API_URL = 'UPLOAD_DIRECTIVE_API_URL';
var ɵ0 = {
    example: 'example'
};
var AngularReactiveFormModule = /** @class */ (function () {
    function AngularReactiveFormModule(injector) {
        setAppInjector(injector);
    }
    AngularReactiveFormModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        ReactiveFormsModule,
                        DirectivesModule,
                        EditorModule
                    ],
                    declarations: [AngularReactiveFormComponent, ReactiveFormComponent, DynamicInputComponent],
                    exports: [AngularReactiveFormComponent, ReactiveFormComponent],
                    providers: [
                        {
                            provide: UPLOAD_DIRECTIVE_HEADERS,
                            useValue: ɵ0
                        },
                        {
                            provide: UPLOAD_DIRECTIVE_API_URL,
                            useValue: 'example'
                        }
                    ]
                },] }
    ];
    /** @nocollapse */
    AngularReactiveFormModule.ctorParameters = function () { return [
        { type: Injector }
    ]; };
    return AngularReactiveFormModule;
}());
export { AngularReactiveFormModule };
export { ɵ0 };
