// @progress/kendo-angular-toolbar/dist/npm/tools/toolbar-buttongroup.component.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var kendo_angular_l10n_1 = require("@progress/kendo-angular-l10n");
var toolbar_tool_component_1 = require("./toolbar-tool.component");
var toolbar_button_component_1 = require("./toolbar-button.component");
var kendo_angular_buttons_1 = require("@progress/kendo-angular-buttons");
var util_1 = require("../util");
/**
 * Represents the Kendo UI Toolbar ButtonGroup for Angular.
 */
var ToolBarButtonGroupComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ToolBarButtonGroupComponent, _super);
    function ToolBarButtonGroupComponent(localization) {
        var _this = _super.call(this) || this;
        _this.localization = localization;
        /**
         * By default, the selection mode of the ButtonGroup is set to `multiple`.
         */
        _this.selection = 'multiple';
        /**
         * Changes the visual appearance by using alternative styling options.
         * The `look` property of the ButtonGroup takes precedence over the `look` property
         * of the individual buttons that are part of the group.
         *
         * The available values are:
         * * `bare`
         * * `flat`
         * * `outline`
         */
        _this.look = 'default';
        _this.focusedIndex = -1;
        _this.getNextKey = util_1.getNextKey(_this.localization.rtl);
        _this.getPrevKey = util_1.getPrevKey(_this.localization.rtl);
        return _this;
    }
    Object.defineProperty(ToolBarButtonGroupComponent.prototype, "buttonElements", {
        get: function () {
            return this.getButtonGroup()
                .buttons.filter(function (b) { return !b.isDisabled; })
                .map(function (b) { return b.element; });
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.onFocus = function () {
        // triggered on tabin
        this.focusedIndex = 0;
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.onNavigate = function (ev) {
        ev.preventDefault();
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.selectedChangeHandler = function (state, button) {
        button.selected = state;
        button.selectedChange.emit(state);
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.onButtonClick = function (ev) {
        this.focusedIndex = this.buttonElements.findIndex(function (be) { return be === ev.target || be.contains(ev.target); });
        this.focusButton(this.focusedIndex, ev);
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.canFocus = function () {
        return !(this.disabled || this.buttonElements.length === 0);
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.focus = function (ev) {
        if (ev === void 0) { ev = {}; }
        var nextKey = this.getNextKey(this.overflows);
        var prevKey = this.getPrevKey(this.overflows);
        this.focusedIndex = util_1.getIndexOfFocused(prevKey, nextKey, this.buttonElements)(ev);
        this.focusButton(this.focusedIndex, ev);
    };
    /**
     * @hidden
     */
    ToolBarButtonGroupComponent.prototype.handleKey = function (ev) {
        var nextKey = this.getNextKey(this.overflows);
        var prevKey = this.getPrevKey(this.overflows);
        var peekAtIndex = util_1.makePeeker(this.buttonElements);
        var isUnmodified = util_1.areEqual(this.focusedIndex);
        this.focusedIndex = util_1.seekFocusedIndex(prevKey, nextKey, peekAtIndex)(this.focusedIndex, ev);
        this.focusButton(this.focusedIndex, ev);
        return !isUnmodified(this.focusedIndex);
    };
    ToolBarButtonGroupComponent.prototype.getButtonGroup = function () {
        return this.overflows ? this.overflowButtonGroup : this.toolbarButtonGroup;
    };
    ToolBarButtonGroupComponent.prototype.focusButton = function (index, ev) {
        // Guard against focusing twice on mousedown.
        if (!ev.type || ev.type === 'focus' || ev.type === 'keydown') {
            this.buttonElements[index].focus();
        }
    };
    ToolBarButtonGroupComponent.decorators = [
        { type: core_1.Component, args: [{
                    exportAs: 'kendoToolBarButtonGroup',
                    // tslint:disable-next-line:no-forward-ref
                    providers: [kendo_angular_l10n_1.LocalizationService, { provide: toolbar_tool_component_1.ToolBarToolComponent, useExisting: core_1.forwardRef(function () { return ToolBarButtonGroupComponent; }) }],
                    selector: 'kendo-toolbar-buttongroup',
                    template: "\n        <ng-template #toolbarTemplate>\n            <kendo-buttongroup\n                #toolbarButtonGroup\n                [tabIndex]=\"-1\"\n                [selection]=\"selection\"\n                [disabled]=\"disabled\"\n                [look]=\"look\"\n                [width]=\"width\"\n                (navigate)=\"onNavigate($event)\"\n                (focus)=\"onFocus()\"\n            >\n                <span\n                    kendoButton\n                    *ngFor=\"let button of buttonComponents\"\n                    [ngStyle]=\"button.style\"\n                    [ngClass]=\"button.className\"\n                    [attr.title]=\"button.title\"\n                    [disabled]=\"button.disabled\"\n                    [togglable]=\"button.togglable\"\n                    [primary]=\"button.primary\"\n                    [selected]=\"button.selected\"\n                    [icon]=\"button.toolbarOptions.icon\"\n                    [iconClass]=\"button.toolbarOptions.iconClass\"\n                    [imageUrl]=\"button.toolbarOptions.imageUrl\"\n                    (click)=\"button.click.emit($event); onButtonClick($event)\"\n                    (selectedChange)=\"selectedChangeHandler($event, button)\"\n                >\n                    {{ button.toolbarOptions.text }}\n                </span>\n            </kendo-buttongroup>\n        </ng-template>\n        <ng-template #popupTemplate>\n            <kendo-buttongroup\n                #overflowButtonGroup\n                class=\"k-overflow-button\"\n                [tabIndex]=\"-1\"\n                [selection]=\"selection\"\n                [disabled]=\"disabled\"\n                [look]=\"look\"\n                [width]=\"width\"\n            >\n                <span\n                    kendoButton\n                    class=\"k-overflow-button\"\n                    *ngFor=\"let button of buttonComponents\"\n                    [ngStyle]=\"button.style\"\n                    [ngClass]=\"button.className\"\n                    [attr.title]=\"button.title\"\n                    [disabled]=\"button.disabled\"\n                    [togglable]=\"button.togglable\"\n                    [primary]=\"button.primary\"\n                    [selected]=\"button.selected\"\n                    [icon]=\"button.overflowOptions.icon\"\n                    [iconClass]=\"button.overflowOptions.iconClass\"\n                    [imageUrl]=\"button.overflowOptions.imageUrl\"\n                    (click)=\"button.click.emit($event); onButtonClick($event)\"\n                    (selectedChange)=\"selectedChangeHandler($event, button)\"\n                >\n                    {{ button.overflowOptions.text }}\n                </span>\n            </kendo-buttongroup>\n        </ng-template>\n    "
                },] },
    ];
    /** @nocollapse */
    ToolBarButtonGroupComponent.ctorParameters = function () { return [
        { type: kendo_angular_l10n_1.LocalizationService }
    ]; };
    ToolBarButtonGroupComponent.propDecorators = {
        disabled: [{ type: core_1.Input }],
        selection: [{ type: core_1.Input }],
        width: [{ type: core_1.Input }],
        look: [{ type: core_1.Input }],
        toolbarTemplate: [{ type: core_1.ViewChild, args: ['toolbarTemplate', { static: true },] }],
        popupTemplate: [{ type: core_1.ViewChild, args: ['popupTemplate', { static: true },] }],
        buttonComponents: [{ type: core_1.ContentChildren, args: [core_1.forwardRef(function () { return toolbar_button_component_1.ToolBarButtonComponent; }),] }],
        toolbarButtonGroup: [{ type: core_1.ViewChild, args: ['toolbarButtonGroup',] }],
        overflowButtonGroup: [{ type: core_1.ViewChild, args: ['overflowButtonGroup',] }]
    };
    return ToolBarButtonGroupComponent;
}(toolbar_tool_component_1.ToolBarToolComponent));
exports.ToolBarButtonGroupComponent = ToolBarButtonGroupComponent;
