// @progress/kendo-angular-pager/dist/npm/pager/pager-next-buttons.component.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var pager_context_service_1 = require("./pager-context.service");
var pager_element_component_1 = require("./pager-element.component");
var kendo_angular_l10n_1 = require("@progress/kendo-angular-l10n");
/**
 * Displays buttons for navigating to the next and to the last page ([see example]({% slug pager_settings %})).
 */
var PagerNextButtonsComponent = /** @class */ (function (_super) {
    tslib_1.__extends(PagerNextButtonsComponent, _super);
    function PagerNextButtonsComponent(localization, pagerContext, cd) {
        return _super.call(this, localization, pagerContext, cd) || this;
    }
    Object.defineProperty(PagerNextButtonsComponent.prototype, "disabled", {
        /**
         * @hidden
         *
         * @readonly
         * @type {boolean}
         * @memberOf PagerNextButtonsComponent
         */
        get: function () {
            return this.currentPage === this.totalPages || !this.total;
        },
        enumerable: true,
        configurable: true
    });
    PagerNextButtonsComponent.prototype.onChanges = function (_a) {
        var total = _a.total, skip = _a.skip, pageSize = _a.pageSize;
        this.total = total;
        this.skip = skip;
        this.pageSize = pageSize;
        this.cd.markForCheck();
    };
    PagerNextButtonsComponent.decorators = [
        { type: core_1.Component, args: [{
                    changeDetection: core_1.ChangeDetectionStrategy.OnPush,
                    selector: 'kendo-datapager-next-buttons',
                    template: "\n        <button\n            type=\"button\"\n            class=\"k-link k-pager-nav\"\n            [class.k-state-disabled]=\"disabled\"\n            [title]=\"textFor('nextPage')\"\n            [attr.aria-label]=\"textFor('nextPage')\"\n            [attr.aria-disabled]=\"disabled ? true : undefined\"\n            (click)=\"currentPage !== totalPages ? changePage(currentPage) : false\">\n            <span class=\"k-icon k-i-arrow-e\" [attr.aria-label]=\"textFor('nextPage')\"></span>\n        </button>\n        <button\n            type=\"button\"\n            class=\"k-link k-pager-nav k-pager-last\"\n            [class.k-state-disabled]=\"disabled\"\n            [title]=\"textFor('lastPage')\"\n            [attr.aria-label]=\"textFor('lastPage')\"\n            [attr.aria-disabled]=\"disabled ? true : undefined\"\n            (click)=\"currentPage !== totalPages ? changePage(totalPages-1) : false\">\n            <span class=\"k-icon k-i-seek-e\" [attr.aria-label]=\"textFor('lastPage')\"></span>\n        </button>\n    "
                },] },
    ];
    /** @nocollapse */
    PagerNextButtonsComponent.ctorParameters = function () { return [
        { type: kendo_angular_l10n_1.LocalizationService },
        { type: pager_context_service_1.PagerContextService },
        { type: core_1.ChangeDetectorRef }
    ]; };
    return PagerNextButtonsComponent;
}(pager_element_component_1.PagerElementComponent));
exports.PagerNextButtonsComponent = PagerNextButtonsComponent;
