// @progress/kendo-angular-sortable/dist/es/draggable.directive.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { Directive, HostBinding, Input, Inject, forwardRef, ElementRef, Renderer2 } from '@angular/core';
import { SortableComponent } from './sortable.component';
/**
 * @hidden
 */
var DraggableDirective = /** @class */ (function () {
    function DraggableDirective(
    /* tslint:disable:no-forward-ref */
    parent, el, renderer) {
        this.parent = parent;
        this.el = el;
        this.renderer = renderer;
    }
    Object.defineProperty(DraggableDirective.prototype, "_focused", {
        get: function () {
            return this.disabled ? false : (this.index === this.parent.activeIndex);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DraggableDirective.prototype, "_disabled", {
        get: function () {
            return this.disabled;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DraggableDirective.prototype, "display", {
        get: function () {
            return this.hidden ? "none" : this._display;
        },
        set: function (display) {
            this._display = display;
        },
        enumerable: true,
        configurable: true
    });
    DraggableDirective.prototype.ngOnInit = function () {
        var nativeElement = this.el.nativeElement;
        this.display = nativeElement.style.display;
        if (nativeElement) { // Remove the inline styles after a few releases of the themes with the style.
            this.renderer.setStyle(nativeElement, 'user-select', 'none');
            this.renderer.setStyle(nativeElement, '-ms-user-select', 'none');
            this.renderer.setStyle(nativeElement, '-moz-user-select', 'none');
            this.renderer.setStyle(nativeElement, '-webkit-user-select', 'none');
        }
    };
    DraggableDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[kendoDraggable]'
                },] },
    ];
    /** @nocollapse */
    DraggableDirective.ctorParameters = function () { return [
        { type: SortableComponent, decorators: [{ type: Inject, args: [forwardRef(function () { return SortableComponent; }),] }] },
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    DraggableDirective.propDecorators = {
        index: [{ type: Input }],
        hidden: [{ type: Input }],
        disabled: [{ type: Input }],
        _focused: [{ type: HostBinding, args: ['class.k-state-focused',] }],
        _disabled: [{ type: HostBinding, args: ['attr.aria-disabled',] }],
        display: [{ type: HostBinding, args: ['style.display',] }]
    };
    return DraggableDirective;
}());
export { DraggableDirective };
