// angular-reactive-form/esm5/lib/directives/single-select2/single-select2.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef, ElementRef, Renderer } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as _ from 'lodash';
var SingleSelect2Component = /** @class */ (function () {
    function SingleSelect2Component(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.defaultValue = null;
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
        this.onTouched = (/**
         * @return {?}
         */
        function () { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    SingleSelect2Component.prototype.writeValue = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        if (val) {
            this.value = val;
        }
    };
    Object.defineProperty(SingleSelect2Component.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.onChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} fn
     * @return {?}
     */
    SingleSelect2Component.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    SingleSelect2Component.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    SingleSelect2Component.prototype.setDisabledState = /**
     * @param {?} isDisabled
     * @return {?}
     */
    function (isDisabled) {
        this.disabled = isDisabled;
    };
    /**
     * @return {?}
     */
    SingleSelect2Component.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.defaultValue = this.value;
        setTimeout((/**
         * @return {?}
         */
        function () {
            _this.actionRender();
        }), 1);
    };
    /**
     * @return {?}
     */
    SingleSelect2Component.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.isClear === true) {
            this.defaultValue = undefined;
            setTimeout((/**
             * @return {?}
             */
            function () {
                $('#single-select2-reactive-form-' + _this.id)
                    .select2()
                    .select2('val', 'ok');
            }), 1);
        }
    };
    /**
     * @param {?} data
     * @return {?}
     */
    SingleSelect2Component.prototype.formatValues = /**
     * @param {?} data
     * @return {?}
     */
    function (data) {
        return data;
    };
    /**
     * @return {?}
     */
    SingleSelect2Component.prototype.actionRender = /**
     * @return {?}
     */
    function () {
        // tslint:disable-next-line:prefer-const
        /** @type {?} */
        var $this = this;
        // tslint:disable-next-line:prefer-const
        /** @type {?} */
        var options = {
            formatResult: this.formatValues,
            formatSelection: this.formatValues
        };
        if (this.placeholder !== undefined && this.placeholder != null) {
            options.placeholder = this.placeholder;
        }
        if (this.isDisabled !== undefined && this.isDisabled != null && this.isDisabled === true) {
            options.disabled = true;
        }
        $('#single-select2-reactive-form-' + this.id).select2(options);
        if (_.isUndefined(this.defaultValue)) {
            /** @type {?} */
            var a_1 = $('#single-select2-reactive-form-' + this.id)
                .select2()
                .select2('val');
            this.value = _.find($this.options, (/**
             * @param {?} o
             * @return {?}
             */
            function (o) {
                return o.value === +a_1;
            }));
        }
        if (!_.isUndefined(this.defaultValue)) {
            $('#single-select2-reactive-form-' + this.id)
                .select2()
                .val(this.value.value)
                .trigger('change');
        }
        $('#single-select2-reactive-form-' + this.id).change((/**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var value = $(this).val();
            $this.value = _.find($this.options, (/**
             * @param {?} o
             * @return {?}
             */
            function (o) {
                return o.value === value;
            }));
        }));
    };
    SingleSelect2Component.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'single-select2',
                    template: "<select id=\"single-select2-reactive-form-{{ id }}\" class=\"form-control\" select2>\n  <option *ngFor=\"let opt of options\" value=\"{{ opt.value }}\" selected=\"{{ opt.selected }}\">{{ opt.label }}</option>\n</select>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return SingleSelect2Component; })),
                            multi: true
                        }
                    ],
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    SingleSelect2Component.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer }
    ]; };
    SingleSelect2Component.propDecorators = {
        placeholder: [{ type: Input }],
        options: [{ type: Input }],
        id: [{ type: Input }],
        _value: [{ type: Input }],
        isClear: [{ type: Input }],
        isDisabled: [{ type: Input }]
    };
    return SingleSelect2Component;
}());
export { SingleSelect2Component };
if (false) {
    /** @type {?} */
    SingleSelect2Component.prototype.placeholder;
    /** @type {?} */
    SingleSelect2Component.prototype.options;
    /** @type {?} */
    SingleSelect2Component.prototype.id;
    /** @type {?} */
    SingleSelect2Component.prototype._value;
    /** @type {?} */
    SingleSelect2Component.prototype.isClear;
    /** @type {?} */
    SingleSelect2Component.prototype.isDisabled;
    /** @type {?} */
    SingleSelect2Component.prototype.defaultValue;
    /**
     * @type {?}
     * @private
     */
    SingleSelect2Component.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    SingleSelect2Component.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    SingleSelect2Component.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    SingleSelect2Component.prototype.element;
    /**
     * @type {?}
     * @private
     */
    SingleSelect2Component.prototype.renderer;
}
